lib/tasks/lumen.rake

```diff
@@ -255,6 +255,29 @@ namespace :lumen do
     end
   end
 
+  desc 'Populate domain_counts table'
+  task populate_domain_count: :environment do
+
+    def give_domains_from_urls(infringing_urls_list)
+      infringing_urls_list.map{ |x| Addressable::URI.parse(x.url).domain }.uniq
+    end
+
+    domain_count_map = {}
+    Notice.find_in_batches do |notices|
```

Comment:
This is going to iterate across over ten million notices. The `find_in_batches` will make each batch manageable, but this job is going to run for a LONG time.

Three things I want to think about:
1. Let's make it possible to specify fewer notices. Maybe we'd like to test this by just running it on 10 notices to see how it works. Maybe we'd just like to parse a manageable number as a cron job at a time when most of our users are asleep.
2. Clearly this is intended to be run only once, but what happens if it isn't? I want this to be idempotent -- processing the same notice twice should not result in twice as many `DomainCount` objects existing. How can we avoid double-counting in the event we run this more than once?
3. How does this dovetail with the NoticeSubmissionFinalizer logic in the other branch? Clearly the intent here is that we run this once to populate the tables, and then we update as new objects come in. However, it will take time to run this, so new objects will come in as we are running it. (This is a possible source of double-counting, actually, so whatever we do to address idempotency will also likely handle this situation.)

I'd love for you to come up with a couple of possible fixes here and pitch me on the best one.